# BMMB852 Assignment 14
### 20241215 Kinscherf

## Makefile usage
This Makefile can be used to simulate an RNA-seq count matrix using the PROPER module, then perform analyses for differential expression and the creation of figures based on that count matrix. The Makefile created for this assignment assumes the installation of the Biostar handbook micromamba stats environment, and the bio code bioinformatics toolbox.

This Makefile contains the following targets, described by the first target, ```usage:```
```bash
usage:
	@echo "simulate    # Simulate an RNA-seq count matrix"
	@echo "edgeR    # Run edgeR using counts.csv and design.csv to find differential expression"
	@echo "evaluate    # Print information comparing results of edgeR using simulated data with known changes in simulated data"
	@echo "pca    # Create a PCA plot of the edgeR results to visualize consistency within groups"
	@echo "heatmap    # Create a heatmap of the edgeR results to visualize patterns of differential expression"
```

The edgeR, PCA, and heatmap targets could also be run with an adequately labeled count matrix from assignment 13/featureCounts.

The Makefile was uploaded at https://github.com/nakinscherf/BMMB852/blob/main/week14/Makefile

## Assignment 14
I initially wanted to use the *Agrobacterium tumefaciens* count matrix from last week's assignment. However, I ran into a slight issue, *i.e.* that the edgeR target produced no significant FDRs.

- Factors: WT pssA 
- Group WT has 3 samples.
- Group pssA has 3 samples.
- Method: glm 
- Input: 5552 rows
- Removed: 5197 rows
- Fitted: 355 rows
- Significant PVal:   10 ( 2.80 %)
- Significant FDRs:    0 ( 0.00 %)

This could relate to a note I made in last week's assignment regarding the grouping of samples from the same "batch." Expression level across batch replicates may simply be too variable to find meaningful differential expression between groups. In the interest of creating graphs, I created simulated data.

The initial simulation gave me:
- Genes with data: 4558 
- Genes that changed: 1000 
- Changes we can detect: 253

Running the targets simulate and then edgeR gave me the following results:
- Method: glm 
- Input: 20000 rows
- Removed: 15420 rows
- Fitted: 4580 rows
- Significant PVal:  403 ( 8.80 %)
- Significant FDRs:  154 ( 3.40 %)

According to these results, there is a discrepancy of ~100 genes with differential expression between groups that we should be able to detect versus those that edgeR actually identified. Fortunately, the evaluation indicated that almost all of the FDRs from the edgeR analysis overlapped with the changed genes known to the simulation:
- 141 found in both
- 112 found only in counts.csv 
- 13 found only in edger.csv

Next, I generated the PCA plot for the edgeR results. The groups appeared to be distinct, which is good.
![pca.pdf](https://github.com/nakinscherf/BMMB852/blob/main/week14/pca.pdf)

I then created the heat map for the edgeR results.
![heatmap.pdf](https://github.com/nakinscherf/BMMB852/blob/main/week14/heatmap.pdf)

Although the handbook describes the normalization, the in-figure legend generated by this legend has no indication such as "z-score" to convey that the "values" are normalized z-scores. I have some beginner-level experience working with graphs in R, but I couldn't figure out where to change the label in the R script to make the heat map. I will need to spend more time looking at the heat map tutorial in the handbook.

I believe it would be appropriate to say that we observed 141 differentially expressed genes (those that were identified as both FDRs by edgeR and as changed in the simulated information). The data for some genes seems more reliable than others; for a consistent difference between groups, I would want one group to be roughly similar shades among the samples, in contrast to the contrasting shade on the other side of the row, for the other group. There are many rows for which this seems to be the case (*e.g.* the top 10 rows on the heat map), but other rows where there will be huge variation within a group (e.g. GENE-15627 and GENE-18833). The variation happened more frequently if the overall trend for that group in that row was upregulation of the gene. Groups showing downregulation in a row were more consistent, which I would think made the data more reliable. However, that makes me wonder if there could be an algorithmic or methodological reason for observing more variability in the upregulated genes.

Scrolling through the edgeR file, it's already evident that there should be a clear difference between groups, as they've been identified as differentially expressed. However, I found it interesting to scroll through the counts file and compare my instinct upon seeing the values with the heat map. My eye was fairly good at picking out the genes that were consistently differentially expressed versus those without enough difference between groups. This seemed very validating to me that the human eye is good at pattern recognition.

This report was uploaded to https://github.com/nakinscherf/BMMB852/blob/main/week14/README_hw14.md