## 20241202 lecture notes
### Kinscherf

- creating the stats environment
    - micromamba env list
    - Istvan wants us to have this installed ASAP so we can get help with issues before trying to do the homework
- salmon_env: make it its own environment instead of installing in bioinfo

```bash
bio code
bash src/steup/init-stats.sh
```

- Istvan uses R(Studio)
    - can run through the command line
- biostarhandbook.com/appbio/methods/stats/#creating-a-custom-environment

- introduction to RNA-seq
    - using DNA fragments to quantify (measure strength) of a biological process
    - methods include ChIP-Seq, RNA-Seq, ATAC-Seq, etc.
    - 2 essential elements
        - where do the DNA fragments come from?
        - are we measuring occupancy or intensity?
    - liorpachter.wordpress.com/seq/
        - list (may or may not be up to date) of seq techniques
    - dealing with superimposition of fragments
        - need a tool to recognize spliced alignments
        - transcripts exist in different abundances; everything always relative to each other, not absolute quantification
    - high-level overview of RNA-seq
        - FASTQ (sequencing data) --> align/classify --> count matrix (genes/transcripts) --> stats method --> results (differential expression)
        - 2 stage process
            - get the count matrix (well standardized, a few methods are available)
            - analyze the count matrix (numberous alternatives)

- count matrix
    - rows (genes) and columns (samples)
    - count corresponds to the number of reads assigned to that transcript/gene
    - some situations where the rows & columns may be different characteristics; need to make sure you know what the rows/columns are in a file as the first thing you do when analyzing it
        - always look at data; eye as good instrument
        - should be aware if there's high variability in a group rather than assuming averages give whole story

- replication
    - simplest approach is pairwise comparison (group 2 vs group 1)
        - very important that everything is being compared to group 2 for reasons I didn't follow
        - fold change = avg group 2 / avg group 1
            - often expressed as log2 scale
    - technical replicates vs biological replicates
        - technical: biological source is the same, testing procedural variability
        - biological: testing biological variability
    - accounting for confounding factors

- the next stage to interpret the function of symbols (differential expression past threshold) --> functional enrichment
- genome-based counting vs transcriptome-based counting

- don't wait until 6 months before graduation/defense to do your stats
- "statistics is a method of salvage"
    - "90% of the time you're going to run it incorrectly the first time around"
    - "90% of RNA-seq results [from 10 years ago?] are made up"

- read biostarhandbook.com/survival-guide.html
    - all stats tests depend on assumptions
